### PR TITLE
Deserialized specification property

### PR DESCRIPTION
**Pull request by rzriyazrz**

### CODE REVIEW

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java [hunk 1]
```diff
@@ -13,11 +20,34 @@ public class InventoryService {
 
     @Autowired
     InventoryRepository inventoryRepository;
-    public Inventory getInventories(int id){
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
```
Comment (Sarathcani999) [inline src/main/java/com/example/spring_security_demo/service/InventoryService.java]: Remove this empty line.

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java [hunk 2]
```diff
@@ -13,11 +20,34 @@ public class InventoryService {
...
-        return inventoryRepository.fetchInventoryDetails(InventoryId);
+
+    public Map<String, Object> getInventoryDetails(Long inventoryId) {
+        Map<String, Object> inventoryDetails = inventoryRepository.fetchInventoryDetails(inventoryId);
+
+        if (inventoryDetails == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Inventory not found");
+        }
+
+        // Create a mutable copy of the inventoryDetails map
```
Comment (Sarathcani999) [inline src/main/java/com/example/spring_security_demo/service/InventoryService.java]: Remove this comment.

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java [hunk 3]
```diff
@@ -13,11 +20,34 @@ public class InventoryService {
...
+        Map<String, Object> mutableInventoryDetails = new HashMap<>(inventoryDetails);
+
+        // Deserialize specifications if it is returned as a string
```
Comment (Sarathcani999) [inline src/main/java/com/example/spring_security_demo/service/InventoryService.java]: Remove this comment.

**Review by Sarathcani999 — CHANGES_REQUESTED** [review]
Please do the necessary changes.

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java (hunk 1 above)
Comment (rzriyazrz) [reply src/main/java/com/example/spring_security_demo/service/InventoryService.java]: fixed.

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java (hunk 2 above)
Comment (rzriyazrz) [reply src/main/java/com/example/spring_security_demo/service/InventoryService.java]: fixed.

File: src/main/java/com/example/spring_security_demo/service/InventoryService.java (hunk 3 above)
Comment (rzriyazrz) [reply src/main/java/com/example/spring_security_demo/service/InventoryService.java]: fixed.

**Review by Sarathcani999 — APPROVED** [review]
Okay.

**Pull request merged**